app/controllers/course/assessment/programming_evaluations_controller.rb

```diff
@@ -0,0 +1,75 @@
+# The programming question evaluations controller.
+#
+# This does *NOT* inherit from the course/assessment controllers because this needs to query across
+# courses and instances, depending on the user's access permissions.
+#
+# There is no access control for the user beyond the course. That is, if an evaluator is granted
+# access to the course as an evaluator, he will be able to see all outstanding jobs and claim any
+# of them.
+class Course::Assessment::ProgrammingEvaluationsController < ApplicationController
+  around_action :unscope_course
+  before_action :load_programming_evaluation, only: [:update_result]
+  around_action :load_and_authorize_pending_programming_evaluation, only: [:allocate]
+  before_action :set_request_format
+  load_and_authorize_resource :programming_evaluations,
+                              class: Course::Assessment::ProgrammingEvaluation.name,
+                              except: [:allocate]
+
+  def index
+    @programming_evaluations = @programming_evaluations.with_language(language_param)
```

Comment:
I haven't decided yet... lol. It's currently a pending spec. I'm not sure if we need an index